sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -302,9 +302,10 @@ export class ServiceBusClient {
       entityPath,
       sessionId,
       {
+        ...this._clientOptions,
```

Comment:
Sometimes we use getOperationOptions(), sometimes not. Any reason to not use it here?

Reply:
(in fact, should that just happen right at the beginning and the only thing we store is an OperationOptionsBase and not ServiceBusClientOptions if that's the only way we use it)?

Reply:
![image](https://user-images.githubusercontent.com/10452642/108441078-38507080-7209-11eb-8f3b-b7a4b7f0acd2.png)
ServiceBusClientOptions contain retryOptions too.
For the places where we were passing the retryOptions, such as Sender and Receiver, I'm passing the whole clientOptions.
For the spots where we need only the OperationOptionsBase, filtering it with `getOperationOptionsBase`.